presto-parser/src/main/java/com/facebook/presto/sql/parser/ParsingException.java

```diff
@@ -39,7 +39,7 @@ public ParsingException(String message)
 
     public ParsingException(String message, NodeLocation nodeLocation)
     {
-        this(message, null, nodeLocation.getLineNumber(), nodeLocation.getColumnNumber());
+        this(message, null, nodeLocation.getLineNumber(), nodeLocation.getColumnNumber() - 1);
```

Comment:
In general I think it's better to change the `getColumnNumber` to be just a getter and change all callers to initialize these classes correctly. But that would be more code change. Maybe add a comment here to explain why this is needed.

Reply:
Added a comment explaining why this is needed.

Reply:
The comment should probably mention that column number recorded is 0-based, thus in getColumnNumber in both NodeLocation and ParsingException we return columnNumber + 1. So we need to revert this back to a 0-based column number to initialize ParsingException. The provided comment is very hard to understand without reading the code so not very useful. Also nits: space after "//".

Reply:
Or if you feel that's too verbose, you can probably refer to the issue in the comment as well.

Reply:
Update the comment to explain that charPositionInLine is 0-based whereas columnNumber is 1-based and why the decrement operation is needed.